source/extensions/filters/network/thrift_proxy/router/router_impl.cc

```diff
@@ -316,6 +316,10 @@ FilterStatus Router::messageEnd() {
 
   upstream_request_->transport_->encodeFrame(transport_buffer, *upstream_request_->metadata_,
                                              upstream_request_buffer_);
+
+  request_size_ += transport_buffer.length();
+  recordClusterScopeHistogram({upstream_rq_size_}, Stats::Histogram::Unit::Bytes, request_size_);
+
```

Comment:
I think we need to reset request_size_ after recording? I think it's possible to reuse a connection for multiple requests, right? (Likewise response_size_).

Reply:
But this is bound to the router filter, not to the connection... won't a new Router filter be created for the next incoming request:

https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/network/thrift_proxy/conn_manager.cc#L181

Reply:
Yeah, that's seems right.